Update release checklists

amyblais opened the pull request:
Can do further iterations + move some items to the new T-5 checklist that will be added via https://github.com/mattermost/docs/pull/3415 in a follow-up PR once these both are merged.

This is to help accountability and that all checklist items are done on time.

jasonblais commented on source/process/bug-fix-release.md:
```diff
@@ -84,12 +85,8 @@ Review the [Release Features & Bugs Quality Gate Guidelines](https://docs.google
-       - Update download links and testing server links to the latest RCs
-    - After build is cut, tweet announcement that RC1 is ready (see [example](https://community.mattermost.com/core/pl/tefx1ijyz7bs8mabuxmpq9f7pw))
     - Generate a list of contributors for Changelog
+    - Ask Platform team to provide list of [database changes](https://github.com/mattermost/mattermost-server/blob/master/store/sqlstore/upgrade.go), [API changes](https://github.com/mattermost/mattermost-server/commits/master/model/client.go), and [WebSocket event changes](https://github.com/mattermost/mattermost-server/blob/master/model/websocket_message.go#L13) for Changelog
```
For platform team regarding database/API/... updates - do you ask a specific dev, dev lead, or the team in general? Do you have an example post to link to?

jasonblais commented on source/process/bug-fix-release.md:
```diff
@@ -106,27 +103,21 @@ Review the [Release Features & Bugs Quality Gate Guidelines](https://docs.google
     - Update Redux before each RC and Final build
     - Master is tagged and branched and `Release Candidate 1` is cut (e.g., 3.5.0-RC1) according to the Release Candidate Checklist in ``mattermost/process``
     - After branching, the database version in `sql_upgrade.go` on master is set to the next scheduled release version (e.g., 3.6.0)
-    - CI servers are updated to the release branch
```
What happens to all these removed items?

Edit: Looks like doc tasks were moved to later in the process. Should there be a step where you ask PMs/tech writing to complete docs by a certain date (which you already do, not sure where it's in the process)?

jasonblais commented on source/process/bug-fix-release.md:
```diff
@@ -137,13 +128,10 @@ Review the [Release Features & Bugs Quality Gate Guidelines](https://docs.google
-    - Review PRs made from release branch and merge changes into the release branch as required and merge the release branch back into master once per day
+    - Review PRs made from release branch and cherry-pick changes into the release branch
     - Run daily automated upgrade tests to avoid catching upgrade bugs late
-6. Build:
```
How do you as the release manager do these tasks?

amyblais commented:
1. Experience. Some items get normally done earlier or later in the process so I moved them accordingly. A few minor items are invalid so I removed them. This way items will get marked as done on time.
2. Done.
3. Hard to add since there's 5 or more different repos and they are different each release.

jasonblais commented:
@amyblais Not merging, as I wasn't sure if you wanted to have anyone else review prior to merge,
